dashboard/test/models/section_test.rb

```diff
@@ -1,69 +1,73 @@
 require 'test_helper'
 
 class SectionTest < ActiveSupport::TestCase
-  test "do not attempt to create sections with duplicate random codes" do
-    teacher = create(:teacher)
+  setup do
+    @student = create :student
+    @teacher = create :teacher
+    @section = create :section, teacher: @teacher
 
-    srand 1
-    s1 = Section.create!(user: teacher, name: "section 1")
+    @default_attrs = {user: @teacher, name: 'test-section'}
+  end
 
-    # seed the RNG with the same thing so we get the same "random" numbers
-    srand 1
-    s2 = Section.create!(user: teacher, name: "section 2")
+  test "create assigns unique section codes" do
+    sections = []
+    3.times do |_i|
+      # Repeatedly seed the RNG so we get the same "random" codes.
+      srand 1
+      sections << Section.create!(@default_attrs)
+    end
```

Comment:
I wonder if this would make more sense to stub `SectionHelpers.random_code`, and `Section.exists?(code: )` as used in [Section.unused_random_code](https://github.com/code-dot-org/code-dot-org/blob/404a93b816491b9808f18f126e3e69680b96b070/dashboard/app/models/section.rb#L167-L168)? Have it return the same code twice, then a different one. Create 2 sections, and expect it's called exactly those 3 times.

Anyway, if you stick with the above loop, it can be simplified as

```ruby
section = 3.times.map do
  # Repeatedly seed the RNG so we get the same "random" codes.
  srand 1
  Section.create!(@default_attrs)
end
```